ingest/spooler.go

```diff
@@ -0,0 +1,467 @@
+	}
+}
+
+func NewS3Spooler(bucket, prefix, region string, mode string, interval time.Duration, stateManager *StateManager, logger *IngestLogger) (*S3Spooler, error) {
+	cfg, err := config.LoadDefaultConfig(context.Background(), config.WithRegion(region))
+	if err != nil {
+		return nil, fmt.Errorf("failed to load AWS config: %w", err)
+	}
+
+	client := s3.NewFromConfig(cfg)
+
+	return &S3Spooler{
+		baseSpooler: &baseSpooler{
+			rowChan:      make(chan SQLiteRow, 1000),
+			stateManager: stateManager,
+			logger:       logger,
+			mode:         mode,
+			interval:     interval,
+		},
+		bucket:    bucket,
+		prefix:    prefix,
+		s3Client:  client,
+		region:    region,
+		awsConfig: cfg,
+	}, nil
+}
+
+func (ls *LocalSpooler) Start(ctx context.Context) error {
+	ls.logger.Info("Starting local spooler in %s mode (directory: %s)", ls.mode, ls.directory)
+
+	go func() {
+		defer close(ls.rowChan)
+
+		for {
+			files, err := ls.discoverFiles()
+			if err != nil {
+				ls.logger.Error("Failed to discover files: %v", err)
+			} else {
+				ls.processFiles(ctx, files)
+			}
+
+			if ls.mode == "once" {
+				ls.logger.Info("Single run complete, exiting spooler")
+				return
+			}
+
+			select {
+			case <-ctx.Done():
+				ls.logger.Info("Context cancelled, stopping spooler")
+				return
+			case <-time.After(ls.interval):
+			}
+		}
+	}()
+
+	return nil
+}
+
+func (ls *LocalSpooler) GetRowChannel() <-chan SQLiteRow {
+	return ls.rowChan
+}
+
+func (ls *LocalSpooler) Stop() error {
+	ls.logger.Info("Stopping local spooler")
+	return nil
+}
+
+func (ls *LocalSpooler) discoverFiles() ([]string, error) {
+	entries, err := os.ReadDir(ls.directory)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read directory: %w", err)
+	}
+
+	var files []string
+	for _, entry := range entries {
+		if entry.IsDir() {
+			continue
+		}
+
+		if !strings.HasSuffix(entry.Name(), ".db.zip") {
+			continue
+		}
+
+		if ls.stateManager.IsProcessed(entry.Name()) {
+			ls.logger.Debug("Skipping already processed file: %s", entry.Name())
+			continue
+		}
+
+		if ls.stateManager.IsFailed(entry.Name()) {
+			ls.logger.Debug("Skipping previously failed file: %s", entry.Name())
+			continue
+		}
+
+		files = append(files, entry.Name())
+	}
+
+	sort.Strings(files)
+	ls.logger.Info("Discovered %d unprocessed files", len(files))
+	return files, nil
+}
+
+func (ls *LocalSpooler) processFiles(ctx context.Context, files []string) {
+	for _, filename := range files {
+		select {
+		case <-ctx.Done():
+			ls.logger.Info("Context cancelled during file processing")
+			return
+		default:
+		}
+
+		filePath := filepath.Join(ls.directory, filename)
+		ls.logger.Info("Processing file: %s", filename)
+
+		if err := ls.processFile(ctx, filePath, filename); err != nil {
+			ls.logger.Error("Failed to process file %s: %v", filename, err)
+			ls.stateManager.MarkFailed(filename, err.Error())
+		} else {
+			// TODO: Move state update to after Elasticsearch indexing is confirmed.
```

Comment:
Does it make sense to create a new story for this? It seems like the kind of thing we might otherwise totally forget, since I doubt this kind of reliability work will be a priority until a little while later.